File: auth-proxy/controller/tweetController.js
const Connector = require('../service/tweet')

module.exports = {
    getTweets: async (req, res) => {
        const Response = await Connector.getTweets()
        console.log(Response);
        res.status(Response.status)
        res.json(Response.data) // Tweets
    },
    getTweet: async (req, res) => {
        const Response = await Connector.getTweet(req.params.id)
        console.log(Response);
        res.status(Response.status)
        res.json(Response.data) // Tweet
    },
    validateTweet: async (req, res) => {
        const Response = await Connector.validateTweet(req.params.id)
        console.log(Response);
        res.status(Response.status)
        res.json(Response.data) // Tweet
    },
    getValidatedTweetsByUser: async (req, res) => {
        const Response = await Connector.getValidatedTweetsByUser(req.query.user)
        console.log(Response);
        res.status(Response.status)
        res.json(Response.data) // Tweets
    },
    getTopHashtags: async (req, res) => {
        const Response = await Connector.getTopHashtags(req.query.limit)
        console.log(Response);
        res.status(Response.status)
        res.json(Response.data) // Hashtags (strings)
    }
}

File: tweets-reader/src/main/java/com/kairosds/tweetsreader/controller/TweetController.java
package com.kairosds.tweetsreader.controller;

import com.kairosds.tweetsreader.model.Tweet;
import io.swagger.annotations.*;

import javax.validation.Valid;
import java.util.List;

@Api
public interface TweetController {

    @ApiOperation(value = "Return the list of filter Tweets")
    @ApiResponses(value = {@ApiResponse(code = 200, message = "Filter Tweets")})
    List<Tweet> getTweets();

    @ApiOperation(value = "Return a filter Tweet")
    @ApiResponses(value = {
            @ApiResponse(code = 200, message = "Filter Tweet"),
            @ApiResponse(code = 404, message = "Tweet not found")})
    Tweet getTweet(
            @ApiParam(required = true) @Valid long id);

    @ApiOperation(value = "Validate a Tweet")
    @ApiResponses(value = {
            @ApiResponse(code = 200, message = "Validated Tweet"),
            @ApiResponse(code = 404, message = "Tweet not found")})
    Tweet validateTweet(
            @ApiParam(required = true) @Valid long id);

    @ApiOperation(value = "Return the list of the validated Tweets by user")
    @ApiResponses(value = {@ApiResponse(code = 200, message = "Validated Tweets")})
    List<Tweet> getValidatedTweetsByUser(
            @ApiParam(required = true) @Valid String user);

    @ApiOperation(value = "Return the list of the most used Hashtags")
    @ApiResponses(value = {@ApiResponse(code = 200, message = "Top Hashtags")})
    List<String> topHashtags(
            @ApiParam() @Valid Integer limit);
}
